src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -0,0 +1,392 @@
+            throw new GATKException("Reads and offset lists have different sizes!");
+        }
+
+        final List<PileupElement> pileup = new LinkedList<>();
+        for (int i = 0; i < reads.size(); i++) {
+            pileup.add(PileupElement.createPileupForReadAndOffset(reads.get(i), offsets.get(i)));
+        }
+
+        return pileup;
+    }
+
+    /**
+     * Helper routine for converting reads and a single offset to a PileupElement list.
+     */
+    private static List<PileupElement> readsOffsets2Pileup(final List<GATKRead> reads, final int offset) {
+        if (reads == null) {
+            throw new GATKException("Illegal null read list in UnifiedReadBackedPileup");
+        }
+        if (offset < 0) {
+            throw new GATKException("Illegal offset < 0 UnifiedReadBackedPileup");
+        }
+
+        final List<PileupElement> pileup = new LinkedList<>();
+        for (final GATKRead read : reads) {
+            pileup.add(PileupElement.createPileupForReadAndOffset(read, offset));
+        }
+
+        return pileup;
+    }
+
+    /**
+     * Make a new pileup consisting of elements of this pileup that satisfy the predicate.
+     */
+    public ReadPileup makeFilteredPileup(final Predicate<PileupElement> filter){
+        return new ReadPileup(loc, pileupElements.stream().filter(filter).collect(Collectors.toList()));
+    }
+
+    /**
+     * Make a new pileup consisting of only reads on the negative strand.
+     */
+    public ReadPileup getPositiveStrandPileup() {
+        return makeFilteredPileup(p -> !p.getRead().isReverseStrand());
+    }
+
+    /**
+     * Make a new pileup consisting of only reads on the negative strand.
+     */
+    public ReadPileup getNegativeStrandPileup() {
+        return makeFilteredPileup(p -> p.getRead().isReverseStrand());
+    }
+
+    /**
+     * Make a new pileup that contains of only bases with quality >= minBaseQ, coming from
+     * reads with mapping qualities >= minMapQ.
+     */
+    public ReadPileup getBaseAndMappingFilteredPileup(final int minBaseQ, final int minMapQ) {
+        return makeFilteredPileup(p -> p.getRead().getMappingQuality() >= minMapQ && (p.isDeletion() || p.getQual() >= minBaseQ));
+    }
+
+    /**
+     * Make a new pileup that contains of only bases with quality >= minBaseQ.
+     */
+    public ReadPileup getBaseFilteredPileup(final int minBaseQ) {
+        return makeFilteredPileup(p -> (p.isDeletion() || p.getQual() >= minBaseQ));
+    }
+
+    /**
+     * Make a new pileup that contains of only bases coming from reads with mapping quality >= minMapQ.
+     */
+    public ReadPileup getMappingFilteredPileup(final int minMapQ) {
+        return makeFilteredPileup(p -> p.getRead().getMappingQuality() >= minMapQ);
+    }
+
+    /**
+     * Makes a new pileup by subsetting reads from a given read group.
+     * Returns null if no reads are from the given read group.
+     * @param targetReadGroupId Identifier for the read group.
```

Comment:
Document how the method behaves if either the `targetReadGroupId` parameter or the read group inside the reads are null.